int8 channel
int32 current # current in mA